### DISCUSSION TITLE
define filters in PHP instead of JS

### DISCUSSION THREAD

**Discussion by olivierbaudouin** [abbasudo/laravel-purity#36] [Q&A] [answered]
I try to define filters in PHP, but I wonder if this PHP syntax is documented.

```
$filters = [...]
User::filter($filters)->get();
```

Why in PHP? Because qs doesn't work yet with Vite bundler.

**Comment by abbasudo** [comment]
achieve this by this approach
```php
$filters = [
      'title' => ['$eq' => 'good post']
];

User::filter($filters)->get();
```

read more at this [section in documents](https://github.com/abbasudo/laravel-purity?tab=readme-ov-file#changing-params-source)

**Reply by olivierbaudouin** [reply]
Yes, but for complex filtering with $or / $and?

**Reply by abbasudo** [reply, marked as answer]
For this task, you can use nested arrays
```php
$filters = [
      '$or' => [
          'title' => ['$eq' => 'good post'],
          'id'    => ['$eq' => '1']
       ]
];
```
